models/git_diff.go

```diff
@@ -225,11 +241,167 @@ type Diff struct {
 	IsIncomplete                 bool
 }
 
+// LoadComments loads comments into each line
+func (diff *Diff) LoadComments(issue *Issue, currentUser *User) error {
+	allComments, err := FetchCodeComments(issue, currentUser)
+	if err != nil {
+		return err
+	}
+	for _, file := range diff.Files {
+		if lineCommits, ok := allComments[file.Name]; ok {
+			for _, section := range file.Sections {
+				for _, line := range section.Lines {
+					if comments, ok := lineCommits[int64(line.LeftIdx*-1)]; ok {
```

Comment:
I find the next lines a bit confusing, might want to add a comment.

* Why is LeftIDX multuiplied by -1?
* Why is the LeftIdx comment replaced an the RightIdx one appended?